### ISSUE TITLE
`default export` should be a compile error

### ISSUE THREAD

**Issue by ubershmekel** [Microsoft/TypeScript#12654]
This feature is called "default export" which caused my weak brain to spell it out that way in my code.

**TypeScript Version:**  2.1.1 / nightly (2.2.0-dev.201xxxxx)

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
var thing = 123;
default export thing;
```

**Expected behavior:**
Compiler error: `Did you mean "export default"?`

**Actual behavior:**
Cryptic runtime failure to use module.

```
C:\myapp\node_modules\express\lib\router\index.js:458
      throw new TypeError('Router.use() requires middleware function but got a ' + gettype(fn));

TypeError: Router.use() requires middleware function but got a undefined
    at Function.use (C:\myapp\node_modules\express\lib\router\index.js:458:13)
    at Function.<anonymous> (C:\myapp\node_modules\express\lib\application.js:219:21)
    at Array.forEach (native)
    at Function.use (C:\myapp\node_modules\express\lib\application.js:216:7)
    at Object.<anonymous> (C:\myapp\app.js:22:5)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
```

**Comment by aluanhaddad** [comment]
![image](https://cloud.githubusercontent.com/assets/5034656/20870632/8b532460-ba59-11e6-9b37-ca74c3b2f285.png)
Not the best error message, but certainly a compile time error (I am using the latest nightly)

**Comment by ubershmekel** [comment]
Thank you @aluanhaddad  I retested now and saw I get that error too. I seem to be experiencing the typescript service failing from time to time. I'll close this ticket and open a different one if the issues continue. I will paste one of the outputs in case I have to come back to it.

```
https://github.com/Microsoft/vscode/issues/12390

[Error - 2:55:55 PM] 'completions' request failed with error.
Error processing request. Cannot read property 'languageServiceRefCount' of undefined
TypeError: Cannot read property 'languageServiceRefCount' of undefined
    at Object.releaseDocumentWithKey (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:46382:18)
    at synchronizeHostData (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:47025:42)
    at Object.getProgram (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:47128:13)
    at ConfiguredProject.Project.updateGraphWorker (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:52593:53)
    at ConfiguredProject.Project.updateGraph (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:52573:39)
    at ConfiguredProject.Project.getLanguageService (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:52404:26)
    at IOSession.Session.getCompletions (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:54974:43)
    at Session.handlers.ts.createMap._a.(anonymous function) (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:54219:61)
    at IOSession.Session.executeCommand (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:55286:28)
    at IOSession.Session.onMessage (c:\Program Files (x86)\Microsoft VS Code\resources\app\extensions\typescript\node_modules\typescript\lib\tsserver.js:55306:35)
```

**Comment by mhegazy** [comment]
@ubershmekel can you share repro steps to reproduce this error you reported? what version of VSCode and what version of TS are you using?

**Comment by ubershmekel** [comment]
@mhegazy I'm not sure about the repro steps. But before I knew about `"watch": true` I was using `Ctrl-Shift-B` to build. Then when I turned on `watch` I would still sometimes try to build though I would get a `Warn There is an active running task right now. Terminate it first before executing another task.` toast, to which I would respond with `Terminate Running Task` sometimes, then my `watch` would die until the next time I hit `Ctrl-Shift-B`. I can't say for certain if that's how I got that output. Sorry.

VSCode 1.7.2
tsc Version 2.0.10
